homeassistant/components/altruist/sensor.py

```diff
@@ -0,0 +1,245 @@
+        key="SHT3X_humidity",
+        translation_key="sht_humidity",
+        native_unit_of_measurement=PERCENTAGE,
+        suggested_display_precision=2,
+    ),
+    "SHT3X_temperature": AltruistSensorEntityDescription(
+        device_class=SensorDeviceClass.TEMPERATURE,
+        key="SHT3X_temperature",
+        translation_key="sht_temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        suggested_display_precision=2,
+    ),
+    "signal": AltruistSensorEntityDescription(
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        key="signal",
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        suggested_display_precision=0,
+    ),
+    "PCBA_noiseMax": AltruistSensorEntityDescription(
+        device_class=SensorDeviceClass.SOUND_PRESSURE,
+        key="PCBA_noiseMax",
+        translation_key="noise_max",
+        native_unit_of_measurement=UnitOfSoundPressure.DECIBEL,
+        suggested_display_precision=0,
+    ),
+    "PCBA_noiseAvg": AltruistSensorEntityDescription(
+        device_class=SensorDeviceClass.SOUND_PRESSURE,
+        key="PCBA_noiseAvg",
+        translation_key="noise_avg",
+        native_unit_of_measurement=UnitOfSoundPressure.DECIBEL,
+        suggested_display_precision=0,
+    ),
+    "CCS_CO2": AltruistSensorEntityDescription(
+        device_class=SensorDeviceClass.CO2,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        translation_key="ccs_co2",
+        key="CCS_CO2",
+        suggested_display_precision=2,
+    ),
+    "CCS_TVOC": AltruistSensorEntityDescription(
+        device_class=SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS,
+        key="CCS_TVOC",
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        suggested_display_precision=2,
+    ),
+    "GC": AltruistSensorEntityDescription(
+        key="GC",
+        native_unit_of_measurement="μR/h",
+        translation_key="radiation",
+        suggested_display_precision=2,
+    ),
+    "SCD4x_co2": AltruistSensorEntityDescription(
+        device_class=SensorDeviceClass.CO2,
+        translation_key="scd4x_co2",
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        key="SCD4x_co2",
+        suggested_display_precision=2,
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: AltruistConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Add sensors for passed config_entry in HA."""
+    coordinator: AltruistDataUpdateCoordinator = config_entry.runtime_data
+    if coordinator.client is not None:
+        async_add_entities(
+            AltruistSensor(
+                coordinator, coordinator.client.device, SENSOR_DESCRIPTIONS[sensor_name]
+            )
+            for sensor_name in coordinator.client.sensor_names
+            if sensor_name in SENSOR_DESCRIPTIONS
```

Comment:
what if we just do
```suggestion
            AltruistSensor(
                coordinator, coordinator.client.device, sensor_description
            )
           for sensor_description in SENSOR_DESCRIPTIONS
           if sensor_description.key in coordinator.client.sensor_names    
```
that should save us the duplicate key

Reply:
while we're at it: Why pass the device as parameter, if you can just get it from the coordinator